source/rofi-icon-fetcher.c

```diff
@@ -282,6 +451,64 @@ gboolean rofi_icon_fetcher_file_is_image(const char *const path) {
   return FALSE;
 }
 
+// build thumbnail's path using md5 hash of an entry name
+static gchar* rofi_icon_fetcher_get_thumbnail(gchar* name, 
+                                              int requested_size, 
+                                              int *thumb_size) {
+  // calc entry_name md5 hash
+  GChecksum* checksum = g_checksum_new(G_CHECKSUM_MD5);
+  g_checksum_update(checksum, (guchar*)name, -1);
+  const gchar *md5_hex = g_checksum_get_string(checksum);
+
+  // determine thumbnail folder based on the request size
+  const gchar* cache_dir = g_get_user_cache_dir();
+  gchar* thumb_path;
+
+  if (requested_size <= 128) {
+    *thumb_size = 128;
+    thumb_path = g_strconcat(cache_dir, "/thumbnails/normal/",
+        md5_hex, ".png", NULL);
+  } else if (requested_size <= 256) {
+    *thumb_size = 256;
+    thumb_path = g_strconcat(cache_dir, "/thumbnails/large/",
+        md5_hex, ".png", NULL);
+  } else if (requested_size <= 512) {
+    *thumb_size = 512;
+    thumb_path = g_strconcat(cache_dir, "/thumbnails/x-large/",
+        md5_hex, ".png", NULL);
+  } else {
+    *thumb_size = 1024;
+    thumb_path = g_strconcat(cache_dir, "/thumbnails/xx-large/",
```

Comment:
I did not have the `thumbnails/xx-large` directory on my system for some reason which blocked the generation of large previews, at least with `xapp-mp3-thumbnailer`.

We can create the target directory if it does not exist before running the thumbnailing commands.